vulnerabilities/data_source.py

```diff
@@ -214,104 +212,42 @@ def __post_init__(self):
             logger.warn(f"AdvisoryData with no tzinfo: {self!r}")
 
 
-class InvalidConfigurationError(Exception):
-    pass
-
-
-@dataclasses.dataclass
-class DataSourceConfiguration:
+class NoLicenseWarning(Warning):
     pass
 
 
-class DataSource(ContextManager):
+class DataSource:
     """
-    This class defines how importers consume advisories from a data source.
-
-    It makes a distinction between newly added records since the last run and modified records. This
-    allows the import logic to pick appropriate database operations.
+    A DataSource collects data from various upstreams and returns corresponding
+    AdvisoryData objects in its advisory_data method.
+    Subclass this class to implement an importer
     """
 
-    CONFIG_CLASS = DataSourceConfiguration
+    license = ""
 
-    def __init__(
-        self,
-        last_run_date: Optional[datetime.datetime] = None,
-        cutoff_date: Optional[datetime.datetime] = None,
-        config: Optional[Mapping[str, Any]] = None,
-    ):
-        """
-        Create a DataSource instance.
-
-        :param last_run_date: Optional timestamp when this data source was last inspected
-        :param cutoff_date: Optional timestamp, records older than this will be ignored
-        :param config: Optional dictionary with subclass-specific configuration
-        """
-        config = config or {}
-        try:
-            self.config = self.__class__.CONFIG_CLASS(**config)
-            # These really should be declared in DataSourceConfiguration above but that would
-            # prevent DataSource subclasses from declaring mandatory parameters (i.e. positional
-            # arguments)
-            setattr(self.config, "last_run_date", last_run_date)
-            setattr(self.config, "cutoff_date", cutoff_date)
-        except Exception as e:
-            raise InvalidConfigurationError(str(e))
-
-        self.validate_configuration()
-
-    def __enter__(self):
-        pass
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        pass
+    def __init__(self):
+        if not self.license:
+            warnings.warn(f"Running importer {self!r} without a license", NoLicenseWarning)
```

Comment:
What about rather raising an exception and make this mandatory? I am not a big fan of warnings.
```suggestion
            raise Exception(f"Cannot run importer {self!r} without a license")
```

Reply:
@pombredanne I initially made this an exception but couldn't find a license for nginx advisories, that way we'd have no working importers.

Reply:
we'll use `license_expression`